### DISCUSSION TITLE
Force to invalidate cached website

### DISCUSSION THREAD

**Discussion by DominykasZaksas** [vercel/next.js#45436] [Help]
Some years ago I made a personal/portfolio website, I dont really remember what was the stack but something simple
Then 3-4 years ago I moved to Next.js

Problem: 
When I open my website it still shows the old version from 3-4 years back even though I have pushed a number of changes and looks and feels are different on nextjs version
If I open a website on a new machine or in a private browser session it works fine

Question:
Is there a way to force you to recache the website remotely? I get that I can invalidate cache on a client-side through browser settings but I cant tell that to a wider audience (regardless of how small it is)
Maybe there are some nifty domain hosting settings somewhere? On domain provider or Vercel?

**Comment by icyJoseph** [comment]
Hi,

> When I open my website it still shows the old version from 3-4 years back

Is this using Next.js still? Do you remember if you used Service Workers? These are notable for causing complications with versions.

A refresh would make the browser go get the latest entry point for the page, for example index.html, while other resources, if still needed are served from cache, if possible. This is nuanced though, and it can be easily affected by other factors.

Could you explain a bit more of your issue? What headers do you see on your page? Is it always serving it from cache?

**Reply by DominykasZaksas** [reply]
Ok, I found the repo of an older project
```
{
  "name": "dz2",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@types/react-router-dom": "^4.3.5",
    "react": "^16.9.0",
    "react-bootstrap": "^0.32.4",
    "react-dom": "^16.9.0",
    "react-router-dom": "^4.3.1",
    "react-scripts-ts": "^4.0.8",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^0.84.0"
  },
  "scripts": {
    "start": "react-scripts-ts start",
    "build": "react-scripts-ts build",
    "test": "react-scripts-ts test --env=jsdom",
    "eject": "react-scripts-ts eject"
  },
  "devDependencies": {
    "@types/jest": "^23.3.14",
    "@types/node": "^10.14.19",
    "@types/react": "^16.9.3",
    "@types/react-dom": "^16.9.1",
    "typescript": "^3.6.3"
  },
  "browserslist": [
    ">0.2%",
    "not dead"
  ]
}
```
It was hosted on firebase.
Yes, it does have a service worker. Using firefox I've tapped push and unregister service worker for a website and now its pulling the latest Nextjs version
Nextjs does not have a service worker

I could still access old version on some different machines if there is some particular header I need to look for

**Reply by icyJoseph** [reply]
Ugh I am not sure, but service workers are notorious for causing issues:

https://twitter.com/sanketsahu/status/1133223194888773632

I think Kent C Dodds once deployed a malfunctioning service worker and he had to go on a twitter to get people to refresh their browsers, but I think he eventually found some way to tell the servicer worker to destroy itself:

https://twitter.com/kentcdodds/status/1281938429702332416

Which I think was taken from: https://medium.com/@nekrtemplar/self-destroying-serviceworker-73d62921d717 

Do give it a try and let us know how it goes! Too bad Service Workers can play us bad like this 😢
